driver/src/price_finding/naive_solver.rs

```diff
@@ -53,20 +49,26 @@ impl Matchable for Order {
         state: &AccountState,
         fee: &Option<Fee>,
     ) -> Option<OrderPairType> {
-        if self.sufficient_seller_funds(&state)
-            && other.sufficient_seller_funds(&state)
-            && self.attracts(other, fee)
+        if !self.sufficient_seller_funds(&state)
+            || !other.sufficient_seller_funds(&state)
+            || !self.attracts(other, fee)
+            || !fee
```

Comment:
Why did these all become `||` from `&&`?

Reply:
`x && y` <=> `!x || !y`. I just wanted to remove the nested if statements as they were not needed.